sort.go

```diff
@@ -0,0 +1,182 @@
+package parquet
+
+// The SortConfig type carries configuration options used to generate sorting
+// functions.
+//
+// SortConfig implements the SortOption interface so it can be used directly as
+// argument to the SortFuncOf function, for example:
+//
+//	sortFunc := parquet.SortFuncOf(columnType, &parquet.SortConfig{
+//		Descending: true,
+//		NullsFirst: true,
+//	})
+//
+type SortConfig struct {
+	MaxRepetitionLevel int
+	MaxDefinitionLevel int
+	Descending         bool
+	NullsFirst         bool
+}
+
+// Apply applies options to c.
+func (c *SortConfig) Apply(options ...SortOption) {
+	for _, opt := range options {
+		opt.ConfigureSort(c)
+	}
+}
+
+// ConfigureSort satisfies the SortOption interface.
+func (c *SortConfig) ConfigureSort(config *SortConfig) {
```

Comment:
I've made this comment before, but the ergonomics of having a value and then calling a function on it with another value of the same type, only to copy it, just feels very odd... 

Why do you want/need to pass it in? Can't you just have `c` pass a `SortConfig` to the caller?

Like I would maybe name this 

```go
func (c *SortConfig) Clone() *SortConfig {
    
}
```

Reply:
As documented on the `SortConfig` type, the intent of this method is to allow passing the configuration as option to the constructor function, I'm not sure how we could do that with the `Clone` method, can you share more details of how you would do this?

Reply:
To be honest I'd probably just make `SortFuncOf` accept a single argument, SortConfig, that's configured however people want things to be sorted...

Reply:
Maybe we can open an issue to discuss the config management strategy we would prefer to adopt in the library, it's a broad topic with lots of material available both online and in internal code bases.